registry-upload/src/generate.ts

```diff
@@ -0,0 +1,56 @@
+import { exec } from "child_process";
+import { writeFileSync } from "fs";
+import { Octokit } from "octokit";
+import { promisify } from "util";
+import { Stage, getExtensionGitHubRepos } from "./common";
+
+const TESTING_EXTENSION: Extension = { id: "testing-extension", url: "https://d3d6d68cbkri8h.cloudfront.net/testing-extension.js" }
+
+generateAllRegistries().catch((reason) => console.log("Failure:", reason))
+
+
+async function generateAllRegistries() {
+    const extensions = await fetchLatestExtensions(getExtensionGitHubRepos())
+    const testStageExtensions = extensions.concat(TESTING_EXTENSION)
+    writeRegistry(Stage.Prod, { extensions })
+    writeRegistry(Stage.Test, { extensions: testStageExtensions })
+}
+
+interface RegistryContent {
+    extensions: Extension[]
+}
+
+interface Extension {
+    id: string
+    url: string
+}
+
+async function fetchLatestExtensions(gitHubRepos: string[]): Promise<Extension[]> {
+    const octokit = await createGitHubClient()
+    async function fetchLatestExtension(gitHubRepo: string): Promise<Extension> {
+        const latestRelease = await octokit.rest.repos.getLatestRelease({ owner: "exasol", repo: gitHubRepo })
+        const version = latestRelease.data.tag_name
+        const extensionAssetsNames = latestRelease.data.assets.map(asset => asset.name).filter(name => name.endsWith(".js"))
+        if (extensionAssetsNames.length !== 1) {
+            const url = `https://github.com/exasol/${gitHubRepo}/releases/tag/${version}`
+            throw new Error(`Expected exactly one .js extension in release ${url}, but got ${extensionAssetsNames.length}`)
+        }
+        return { id: gitHubRepo, url: `https://extensions-internal.exasol.com/com.exasol/${gitHubRepo}/${version}/${extensionAssetsNames[0]}` }
```

Comment:
Maybe extract to a dedicated method.

Reply:
with a comment about rationale regarding count downloads by customers separate from downloads triggered by employees of Exasol company.